business-registry-dashboard/app/middleware/02.magic-link.global.ts

```diff
@@ -0,0 +1,12 @@
+export default defineNuxtRouteMiddleware((to) => {
+  const localePath = useLocalePath()
+
+  // Check if path contains affiliationInvitation and extract token
+  if (to.path.includes('affiliationInvitation/acceptToken/')) {
+    const token = to.path.split('acceptToken/')[1]
+    if (token) {
+      sessionStorage.setItem('affiliationToken', token)
+    }
+    window.location.href = localePath('/')
```

Comment:
So, if the path contains a token then you save it to session storage and then reload everything?

Reply:
Yep, exactly.

Reply:
Why so complicated? Why can't the token be handled directly?

Reply:
Because any route that is not that the main route will give a page 404 error:
![image](https://github.com/user-attachments/assets/6df30b35-c9bb-43eb-a18c-654a7ab0a7f1)

Whenever that link is pressed, we will get an error if we don't handle it using a middleware or something else. It's not possible to handle it from the `index.vue` file directly as we won't even make it to that file.

When I tried handling that via an actual route, I was getting that error as well since the email returned has a double slash `//`. The browser was standardizing the URL and removing org ID part. It's a weird thing really. That's why I moved away from that design as I tried countless things but couldn't manage to deal with all the edge cases properly.

Reply:
This shouldn't be an issue now

Reply:
I love the new design!

Please resolve this conversation.